src/main/java/com/simibubi/create/content/trains/entity/Train.java

```diff
@@ -620,6 +624,62 @@ public boolean hasBackwardConductor() {
 		return false;
 	}
 
+
+	private void updateCollisionCache() {
+		if (derailed || graph == null) {
+			if (collisionCache != null)
+				collisionCache.invalidate();
```

Comment:
Why do we invalidate and then not update? Leave a comment please

Reply:
Good practice to make it obvious that that's the _intention_ for future you/me/maintainers. I can easily look at the code and decide that's what's happening but I cannot easily decide what you meant to be happening